azure/scope/managedcontrolplane.go

```diff
@@ -413,7 +413,7 @@ func (s *ManagedControlPlaneScope) ManagedClusterAnnotations() map[string]string
 // ManagedClusterSpec returns the managed cluster spec.
 func (s *ManagedControlPlaneScope) ManagedClusterSpec(ctx context.Context) azure.ResourceSpecGetter {
 	managedClusterSpec := managedclusters.ManagedClusterSpec{
-		Name:              s.ControlPlane.Name,
+		Name:              s.ClusterName(),
```

Comment:
@zmalik is this change as-is going to break back-compat for existing clusters?

Reply:
It shouldn't, my understanding from #2438 is that anything that has a mismatch between cluster name and control plane name will currently fail, so there shouldn't be any existing clusters where these two values aren't the same.

Reply:
Just to make sure, should this be the `Cluster` name or the `AzureManagedCluster` name?

Reply:
As far as I could deduce, unless we want the user to be able to configure different names for `AzureManagedCluster` and `Cluster`, we should be good to use `Cluster` name here. 

Because, we seem to have a tight-coupling b/w `Cluster` name and `AzureManagedCluster` name in our implementation. Referring to the published examples:
- https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/templates/cluster-template-aad.yaml
- https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/templates/cluster-template-aks.yaml
- https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/templates/cluster-template.yaml

Moreover, we are borrowing [`ClusterName` as `AzureManagedCluster`'s name in here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/d28ff9a7d44b25fe10ca3e13a7e2725b50b35fad/azure/scope/managedcontrolplane.go#L354) (hinting that `Cluster` name and `AzureManagedCluster` name are interchangeable in our implementation.)

However, 

CAPI provides examples where the user has the freedom to name _cluster_ and underlying infrastructure independently. 
- https://github.com/kubernetes-sigs/cluster-api/blob/main/test/infrastructure/docker/templates/clusterclass-quick-start.yaml#L1-L20

Reply:
In theory, nothing should prevent you to use a different resource name for Cluster and AzureCluster (or AzureManagedCluster in the case of managed clusters) so I think we should account for that.

> [ClusterName as AzureManagedCluster's name in here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/d28ff9a7d44b25fe10ca3e13a7e2725b50b35fad/azure/scope/managedcontrolplane.go#L354)

that seems like a logging bug

Reply:
@nawazkh were you able to repro https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2438 ? I'm having a hard time seeing what causes the "not found" error since all the code paths you pointed out above in PR description are using the same name (AzureManagedControlPlane's name) and nothing seems to be using AzureManagedCluster, so wondering where the inconsistency is coming from.

Reply:
Paired with @CecileRobertMichon to investigate when #2438 issue was not showing up on _main_ branch. We found that #2438 was fixed as part of PR https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2168.
Will advise the user to use latest version of CAPZ.

Thank you Cecile!